docs/adding_packages/conandata_yml_format.md

```diff
@@ -146,98 +135,3 @@ patches:
       patch_type: "portability"
       patch_source: "https://a-url-to-a-pull-request-mail-list-topic-issue-or-question"
 ```
-
```

Comment:
why do you delete this chunk of the documentation?
it doesn't seem to be related to linter at all

Reply:
These are not required and not enforced + the "rules" are duplicate of https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/sources_and_patches.md#rules so it seemed appropriate since they were no longer referenced

Reply:
these are not "rules", it's a documentation/reference on our `conandata.yml` format. which custom fields it has, which values are allowed and so on. it should be left.
in general, "delete linter" PR shouldn't touch unrelated parts of the documentation. in other words, change should be atomic and isolated.